screen2.0/src/app/downloads/datamatrices.tsx

```diff
@@ -99,32 +125,56 @@ export function DataMatrices() {
   const [lifeStage, setLifeStage] = useState("all")
   const [colorBy, setColorBy] = useState<"ontology" | "sampleType">("ontology")
   const [tSelected, setTSelected] = useState(new Set([]))
-  const [searched, setSearched] = useState<BiosampleUMAP>(null)
+  const [searched, setSearched] = useState<String>(null)
   const [biosamples, setBiosamples] = useState<BiosampleUMAP[]>([])
   const [selectMode, setSelectMode] = useState<"select" | "zoom">("select")
   const [tooltip, setTooltip] = useState(-1)
-  
-  const data = useMemo(() =>{
-    return umapData && umapData.ccREBiosampleQuery.biosamples.length>0 ? umapData: {} 
-  }, [umapData])
 
-  const [open, setOpen] = useState(false)
-  const handleOpenModal = () => {
-    biosamples.length !== 0 && setOpen(true)
+  const handleSetSelectedSample = (selected: any) => {
+    setSearched(selected.displayname)
   }
-  const handleCloseModal = () => setOpen(false)
+
+  const [openModalType, setOpenModalType] = useState<null | "biosamples" | "download">(null);
+
+  const handleOpenModal = () => {
+    if (biosamples.length !== 0) {
+      setOpenModalType("biosamples");
+    }
+  };
+
+  const handleOpenDownloadModal = () => {
+    setOpenModalType("download");
+  };
+
+  const handleCloseModal = () => {
+    setOpenModalType(null);
+  };
+  
+  const [selectedFormats, setSelectedFormats] = useState({
+    signal: false,
+    zScore: false,
+  });
+  
+  const handleDownload = () => {
```

Comment:
When I select both options for download only the z score matrix file starts downloading